### PR TITLE
Add instructions for manual testing in DG

### PR DESCRIPTION
**Pull request by brein62**
For the features:
- Material tracking
- Roles tracking
- Maximum volunteer count

Use cases:
- Update use cases for eaddv to include role and material tracking
- Add use case for eaddm

Suggestions for how to simplify these features are welcome

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -782,6 +782,90 @@ testers are expected to do more *exploratory* testing.
...
+
+1. Setting maximum volunteer capacity
+
+   1. Prerequisites: 
+      - At least 2 volunteers in the displayed volunteer list
+
+   2. Test case instructions in order (take note of the `vs/` parameter at the end):
+      1. `ecreate n/Food packing r/10 chef r/10 packer sd/23/10/2023 1900 ed/23/10/2023 2200 l/hougang dsc/Packing food for the needy m/100 packets b/100.00 vs/1`
+      2. `eaddv eid/X vid/1` (substitute `X` with the index of the `Food packing` event - e.g. `eaddv eid/1 vid/1`)
+      3. `eaddv eid/X vid/2` (substitute `X` with the index of the `Food packing` event - e.g. `eaddv eid/1 vid/2`)
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:836]: should change `substitute X with the` to `where X is the`

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -782,6 +782,90 @@ testers are expected to do more *exploratory* testing.
...
+      4. `eedit X vs/0` (substitute `X` with the index of the `Food packing` event - e.g. `eaddv 1 vs/0`)      
+      5. `eshow X` (substitute `X` with the index of the `Food packing` event - e.g. `eshow 1`)
+      6. Close the `eshow` window either by using the `Esc` key or clicking the close button on the window.
+      7. `eaddv eid/X vid/2` (substitute `X` with the index of the `Food packing` event - e.g. `eaddv eid/1 vid/2`)<br><br>
+      Expected: 
+         - First `eaddv` command (2) runs successfully, and success message that `New volunteer added to event.` with total volunteer count in the event is displayed.
+         - Second `eaddv` command (3) does not run. Error message that `This event has already reached a maximum of 1 volunteer(s), and is unable to accept any more volunteers` is displayed as the status message.
+         - `eedit` command (4) runs successfully, and success message showing the edited event is displayed.<br>
+           <box type="info" seamless>
+              Since the vs/ parameter is set to 0 (special value), it removes the limit to the number of volunteers added to the event.
```
Comment (zacwong2151) [inline docs/DeveloperGuide.md:846]: damn ok I just learnt about this LOL

**Review by zacwong2151 — APPROVED** [review]
lengthy test cases but I think its fair

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (brein62) [reply docs/DeveloperGuide.md:957]: good idea cos i did this for the subsequent ones and forgot change hahah

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (brein62) [reply docs/DeveloperGuide.md:967]: yessir the more you know

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (brein62) [reply docs/DeveloperGuide.md]: fixed :D

**Review by tiongMax — COMMENTED** [review]
Lmao i did the use cases for eaddv, eremovev, elistv ady, such a coincidence

**Comment by tiongMax** [comment]
> Lmao i did the use cases for eaddv, eremovev, elistv ady, such a coincidence

**Comment by tiongMax** [comment]
Oops accidentally closed it, its reopened now. My bad !

**Review by tiongMax — APPROVED** [review]
Lgtm!

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -916,7 +1037,90 @@ testers are expected to do more *exploratory* testing.
...
+      4. `eshow X` (where `X` is the index of the `More food packing` event - e.g. `eshow 2`)
+      5. Close the `eshow` window either by using the `Esc` key or clicking the close button on the window.
+      6. `eedit X vs/1` (where `X` is the index of the `More food packing` event - e.g. `eedit 2 vs/1`)<br><br>
+      Expected:
+         - Both `eaddv` commands (2, 3) run successfully, and success message that `New volunteer added to event.` with total volunteer count in the event is displayed.
+         - `eshow` command (5) runs successfully. A window opens, with one field showing `Maximum number of volunteers in event: No limit`.
+         - `eedit` command (6) does not run successfully. Error message that `The maximum number of volunteers in an event should not be less than the number of volunteers currently in the event. To remove the cap on the number of volunteers, you can use vs/0` is displayed as the status message.
+
+   4. Test case: `ecreate n/Even more food packing r/10 chef r/10 packer sd/23/10/2023 1900 ed/23/10/2023 2200 l/hougang dsc/Packing food for the needy m/100 packets b/100.00 vs/50`, followed by
```
Comment (ruo-x) [inline docs/DeveloperGuide.md:1118]: shouldnt this be under eshow?

**Review by ruo-x — CHANGES_REQUESTED** [review]
v detailed instructions but dk if the last test case considered part of eshow

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -757,14 +781,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+**Use case UCEV01: Add volunteer to an event**
 
 **MSS**
 
-1. User clears all volunteers in volunteer list.
-2. The volunteer list becomes empty.
+1. User <u>creates an event (UCE01)</u> with a certain required quantity of roles.
+2. User <u>creates a volunteer (UCV01)</u> with a certain skill.
+3. User requests to add the volunteer to the event.
+4. Volunteer is added to the Event's volunteer list.
```
Comment (jordankanghm) [inline docs/DeveloperGuide.md:791]: shld this be "iVolunter adds the volunteer to..." to simulate user and system interaction?

File: docs/DeveloperGuide.md [hunk 5]
```diff
@@ -757,14 +781,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+      Use case ends.
+
+**Use case UCEV02: Remove volunteer from an event**
+
+**MSS**
+
+1. User <u>lists all volunteering events (UCE02)</u>.
+2. User <u>lists all volunteers (UCV02)</u>.
+3. User requests to remove a volunteer from the event.
+4. Volunteer is removed from the Event's volunteer list.
```
Comment (jordankanghm) [inline docs/DeveloperGuide.md:830]: same here, maybe can include iVolunteer in some manner?

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -757,14 +781,111 @@ Priorities: High (must have) - `* * *`, Medium (nice to have) - `* *`, Low (unli
...
+    * 3a1. The volunteer list displayed is empty.
+      Use case ends.
+
+**Use case UCEV04: List all events joined by a volunteer**
+
+**MSS**
+
+1. User <u>lists all volunteers (UCV02)</u>.
+2. User requests to list all events joined by a certain volunteer.
+3. iVolunteer lists all event joined by the volunteer in the event list.
```
Comment (jordankanghm) [inline docs/DeveloperGuide.md:878]: events!

**Review by jordankanghm — CHANGES_REQUESTED** [review]
just some minor comments but lgtm overall!

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (brein62) [reply docs/DeveloperGuide.md:1118]: this reflects the `eshow` feature capability to show max volunteers. u can add to ur `eshow` test cases then i'll adjust the test case accordingly

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (brein62) [reply docs/DeveloperGuide.md:791]: will fix

File: docs/DeveloperGuide.md (hunk 6 above)
Comment (brein62) [reply docs/DeveloperGuide.md:878]: i like to list all event too bad

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (brein62) [reply docs/DeveloperGuide.md:1118]: hmm i saw your `eshow` implementation and i think it's fine as long as it's listed after `eshow`. else the same issue would apply for `ecreate` etc.

**Comment by tiongMax** [comment]
Yo guys pls merge this after #229 is merged

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (ruo-x) [reply docs/DeveloperGuide.md:1118]: ok suree we can leave it as it is bah :)

Comment (brein62) [reply docs/DeveloperGuide.md:1118]: yay i'm so happy :D :D :D :D :D :D :D :D

Comment (ruo-x) [reply docs/DeveloperGuide.md:1118]: ?? JAJAHAHA

**Review by ruo-x — APPROVED** [review]
lgtm!!

**Pull request merged**